### PR TITLE
UI tweaks and nits for starter packs

### PR DESCRIPTION
**Pull request by haileyok**

### CODE REVIEW

File: __tests__/lib/string.test.ts [hunk 1]
```diff
@@ -852,11 +856,15 @@ describe('parseStarterPackHttpUri', () => {
...
+    {name: 'did:123', rkey: 'rkey'},
+    null,
+    null,
+    null,
   ]
 
   it('returns the correct name and rkey when input is valid', () => {
     for (let i = 0; i < inputs.length; i++) {
-      const result = parseStarterPackHttpUri(inputs[i])
+      const result = parseStarterPackUri(inputs[i])
```
Comment (estrattonbailey) [inline __tests__/lib/string.test.ts:867]: Kinda nit kinda not: should really use `it.each` and a nice interpolated string for this, or refactor into separate test cases like I did for `ago` tests. Reason being, it's hard to know which part failed in tests like this, why tests exist, if duplicate tests exist, etc, esp when the arrays of tests grow.

Comment (haileyok) [reply __tests__/lib/string.test.ts:867]: Yea, I've been wanting to refactor a couple of the other tests in this thing for the same reason (that embed test is brutal). Might as clean this one up now.

File: __tests__/lib/string.test.ts [hunk 2]
```diff
@@ -856,13 +856,13 @@ describe('createStarterPackLinkFromAndroidReferrer', () => {
...
       createStarterPackLinkFromAndroidReferrer(
-        'utm_content=starterpack-haileyok.com-rkey-more',
+        'utm_content=starterpack_haileyok.com_rkey_more',
       ),
     ).toEqual(null)
 
     expect(
       createStarterPackLinkFromAndroidReferrer(
-        'utm_content=notastarterpack-haileyok.com-rkey',
+        'utm_content=notastarterpack_haileyok.com_rkey',
```
Comment (estrattonbailey) [inline __tests__/lib/string.test.ts:865]: Should add a test with a hyphen in the handle

File: src/lib/generate-starterpack.ts [hunk 3]
```diff
@@ -111,7 +109,20 @@ export async function generateStarterpack({
       )
     ).uri
   } catch (e: unknown) {
-    logger.error('Failed to generate starter pack', {error: e})
+    logger.error('Failed to generate starter pack', {safeMessage: `${e}`})
```
Comment (estrattonbailey) [inline src/lib/generate-starterpack.ts:112]: Doesn't matter but FYI you actually don't need to coerce here, we check for `instanceof Error` in the breadcrumbs handling and call `Error.toString` https://github.com/bluesky-social/social-app/blob/main/src/logger/index.ts#L97

File: __tests__/lib/string.test.ts (hunk 2 above)
Comment (estrattonbailey) [reply __tests__/lib/string.test.ts:871]: Ope nvm I see you switched to underscores

File: src/lib/generate-starterpack.ts (hunk 3 above)
Comment (haileyok) [reply src/lib/generate-starterpack.ts:112]: ah wasn't sure if we did it for each value or not, thanks.

File: src/screens/Onboarding/StepFinished.tsx [hunk 4]
```diff
@@ -72,8 +73,13 @@ export function StepFinished() {
           })
           listItems = listRes.data.items
         }
+      } catch (e) {
+        logger.error('Failed to fetch starter pack', {safeMessage: `${e}`})
+        // don't tell the user, just get them through onboarding.
```
Comment (estrattonbailey) [inline src/screens/Onboarding/StepFinished.tsx:78]: Niiice

**Review by estrattonbailey — APPROVED** [review]

**Pull request merged**